code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm

```diff
@@ -131,21 +143,42 @@
 
 /datum/eldritch_knowledge/final/rust_final
 	name = "Rustbringer's Oath"
-	desc = "Bring 3 corpses onto the transmutation rune. After you finish the ritual rust will now automatically spread from the rune. Your healing on rust is also tripled, while you become more resillient overall."
+	desc = "Bring 3 corpses onto the transmutation rune. After you finish the ritual rust will now automatically spread from the rune. Your healing on rust is also tripled, while you become extremely more resillient."
 	gain_text = "Champion of rust. Corruptor of steel. Fear the dark for the Rustbringer has come! Rusted Hills, CALL MY NAME!"
 	cost = 3
 	required_atoms = list(/mob/living/carbon/human)
 	route = PATH_RUST
+	var/list/conditional_immunities = list(TRAIT_STUNIMMUNE,TRAIT_SLEEPIMMUNE,TRAIT_PUSHIMMUNE,TRAIT_SHOCKIMMUNE,TRAIT_NOSLIPALL,TRAIT_RADIMMUNE,TRAIT_RESISTHIGHPRESSURE,TRAIT_RESISTLOWPRESSURE,TRAIT_RESISTCOLD,TRAIT_RESISTHEAT,TRAIT_PIERCEIMMUNE,TRAIT_BOMBIMMUNE,TRAIT_NOBREATH)
+	///if this is set to true then immunities are active, if false then they are not active, simple as.
+	var/immunities_active = FALSE
 
 /datum/eldritch_knowledge/final/rust_final/on_finished_recipe(mob/living/user, list/atoms, loc)
 	var/mob/living/carbon/human/H = user
 	H.physiology.brute_mod *= 0.5
 	H.physiology.burn_mod *= 0.5
 	H.client?.give_award(/datum/award/achievement/misc/rust_ascension, H)
+	RegisterSignal(H,COMSIG_MOVABLE_MOVED,.proc/on_move)
 	priority_announce("$^@&#*$^@(#&$(@&#^$&#^@# Fear the decay, for the Rustbringer, [user.real_name] has ascended! None shall escape the corrosion! $^@&#*$^@(#&$(@&#^$&#^@#","#$^@&#*$^@(#&$(@&#^$&#^@#", ANNOUNCER_SPANOMALIES)
 	new /datum/rust_spread(loc)
 	return ..()
 
+/datum/eldritch_knowledge/final/rust_final/proc/on_move(mob/mover)
+	SIGNAL_HANDLER
+	var/istype_check = istype(get_turf(mover),/turf/open/floor/plating/rust)
```

Comment:
This ISN'T more descriptive, I don't know what I'm not being clear about.

This is checking something SPECIFICALLY. It is checking if you are on rust. `boolean` is a bad name for this, `expression` is a bad name for this, and `istype_check` is a bad name for this. The name I gave originally, `mover_on_rust` if I remember right, is the *perfect* name for this--it's checking if the mover is on rust. With your current name, it is not obvious unless you refer all the way back to the variable declaration what this actually *is*.

Reply:
done

Reply:
hello @Mothblocks this is finished.